From 1e6a1b7d6b8b8838666538bf5cfa91ebf9d6cde0 Mon Sep 17 00:00:00 2001
From: Julien Cristau <jcristau@debian.org>
Date: Sun, 20 Apr 2008 20:36:44 +0200
Subject: [PATCH] Minor xorg.conf manpage fixups

Use __libmansuffix__ instead of __oslibmansuffix__ which isn't getting
replaced, and rewrap some text to get __xservername__ replaced in the
description of Option "Accel" (cpp doesn't like the preceding quote).
(cherry picked from commit 40c6be1408a1f0b236fdb28af27ae18aea0d578f)
---
 hw/xfree86/doc/man/xorg.conf.man.pre |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/hw/xfree86/doc/man/xorg.conf.man.pre b/hw/xfree86/doc/man/xorg.conf.man.pre
index de93aaf..198a37c 100644
--- a/hw/xfree86/doc/man/xorg.conf.man.pre
+++ b/hw/xfree86/doc/man/xorg.conf.man.pre
@@ -54,7 +54,7 @@ is the relative path (with no \(lq..\(rq components) specified by that
 environment variable, and
 .I <hostname>
 is the machine's hostname as reported by
-.BR gethostname (__oslibmansuffix__).
+.BR gethostname (__libmansuffix__).
 .PP
 When the __xservername__ server is started by the \(lqroot\(rq user, the config file
 search locations are as follows:
@@ -93,7 +93,7 @@ is the path specified by that environment variable (usually the home
 directory), and
 .I <hostname>
 is the machine's hostname as reported by
-.BR gethostname (__oslibmansuffix__).
+.BR gethostname (__libmansuffix__).
 .PP
 The
 .I __xconfigfile__
@@ -1626,8 +1626,8 @@ Others are driver\-independent, and will eventually be described here.
 .\" XXX These should really be in an xaa man page.
 .TP 7
 .BI "Option \*qAccel\*q"
-Enables XAA (X Acceleration Architecture), a mechanism that makes video
-cards' 2D hardware acceleration available to the  __xservername__ server.
+Enables XAA (X Acceleration Architecture), a mechanism that makes video cards'
+2D hardware acceleration available to the  __xservername__ server.
 This option is on by default, but it may be necessary to turn it off if
 there are bugs in the driver.
 There are many options to disable specific accelerated operations, listed
-- 
1.5.6.5

From 754300e87afc2dc23d285f486efb50610f51f5a1 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Thu, 2 Oct 2008 17:56:31 -0400
Subject: [PATCH] xorg.conf.man: We don't look in root's ~ anymore, don't say we do.
 (cherry picked from commit 56a24cf894a142b18c7f18e294074c32031f3d14)

---
 hw/xfree86/doc/man/xorg.conf.man.pre |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/hw/xfree86/doc/man/xorg.conf.man.pre b/hw/xfree86/doc/man/xorg.conf.man.pre
index 198a37c..bd426c4 100644
--- a/hw/xfree86/doc/man/xorg.conf.man.pre
+++ b/hw/xfree86/doc/man/xorg.conf.man.pre
@@ -67,7 +67,6 @@ search locations are as follows:
 .B $XORGCONFIG
 .IB /etc/X11/ $XORGCONFIG
 .IB __projectroot__/etc/X11/ $XORGCONFIG
-.BI $HOME /__xconfigfile__
 .I /etc/X11/__xconfigfile__\-4
 .I /etc/X11/__xconfigfile__
 .I /etc/__xconfigfile__
-- 
1.5.6.5

From aaaf9869e2e72debfbc9565f159951498ce9b82b Mon Sep 17 00:00:00 2001
From: Julien Cristau <jcristau@debian.org>
Date: Sat, 25 Oct 2008 22:54:15 +0200
Subject: [PATCH] Bug#5851: Document the "Log" option in xorg.conf(5)
 (cherry picked from commit fd97ff1bdd30c358525068a4c833f681e00719a2)

---
 hw/xfree86/doc/man/xorg.conf.man.pre |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/hw/xfree86/doc/man/xorg.conf.man.pre b/hw/xfree86/doc/man/xorg.conf.man.pre
index bd426c4..7542aa0 100644
--- a/hw/xfree86/doc/man/xorg.conf.man.pre
+++ b/hw/xfree86/doc/man/xorg.conf.man.pre
@@ -708,6 +708,15 @@ If this option is disabled, then the devices will be added (and the
 DevicePresenceNotify event sent), but not enabled, thus leaving policy up
 to the client.
 Enabled by default.
+.TP 7
+.BI "Option \*qLog\*q \*q" string \*q
+This option controls whether the log is flushed and/or synced to disk after
+each message.
+Possible values are
+.B flush
+or
+.BR sync .
+Unset by default.
 .SH "MODULE SECTION"
 The
 .B Module
-- 
1.5.6.5

From 53dfdff25b2b5d129dc60aa18ef5f3e81e181f1a Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@sun.com>
Date: Mon, 1 Dec 2008 16:28:45 -0800
Subject: [PATCH] Update See Also lists in Xorg & xorg.conf man pages

Remove xorgconfig & xorgcfg
Update driver lists:
	- Remove vga
	- i810->intel
	- via->openchrome
(cherry picked from commit 24c562f04b41d219c34f5fa3f963564accf329f2)
---
 hw/xfree86/doc/man/Xorg.man.pre      |    6 +++---
 hw/xfree86/doc/man/xorg.conf.man.pre |    7 +++----
 2 files changed, 6 insertions(+), 7 deletions(-)

diff --git a/hw/xfree86/doc/man/Xorg.man.pre b/hw/xfree86/doc/man/Xorg.man.pre
index 8b48951..e427e51 100644
--- a/hw/xfree86/doc/man/Xorg.man.pre
+++ b/hw/xfree86/doc/man/Xorg.man.pre
@@ -577,7 +577,7 @@ Initial access control list for display
 .IR n .
 .SH "SEE ALSO"
 X(__miscmansuffix__), Xserver(__appmansuffix__), xdm(__appmansuffix__), xinit(__appmansuffix__),
-__xconfigfile__(__filemansuffix__), xorgconfig(__appmansuffix__), xorgcfg(__appmansuffix__), xvidtune(__appmansuffix__),
+__xconfigfile__(__filemansuffix__), xvidtune(__appmansuffix__),
 apm(__drivermansuffix__),
 ati(__drivermansuffix__),
 chips(__drivermansuffix__),
@@ -588,12 +588,13 @@ glide(__drivermansuffix__),
 glint(__drivermansuffix__),
 i128(__drivermansuffix__),
 i740(__drivermansuffix__),
-i810(__drivermansuffix__),
 imstt(__drivermansuffix__),
+intel(__drivermansuffix__),
 mga(__drivermansuffix__),
 neomagic(__drivermansuffix__),
 nsc(__drivermansuffix__),
 nv(__drivermansuffix__),
+openchrome (__drivermansuffix__),
 r128(__drivermansuffix__),
 rendition(__drivermansuffix__),
 s3virge(__drivermansuffix__),
@@ -612,7 +613,6 @@ trident(__drivermansuffix__),
 tseng(__drivermansuffix__),
 v4l(__drivermansuffix__),
 vesa(__drivermansuffix__),
-vga(__drivermansuffix__),
 vmware(__drivermansuffix__),
 .br
 Web site
diff --git a/hw/xfree86/doc/man/xorg.conf.man.pre b/hw/xfree86/doc/man/xorg.conf.man.pre
index 7542aa0..a3ab9a0 100644
--- a/hw/xfree86/doc/man/xorg.conf.man.pre
+++ b/hw/xfree86/doc/man/xorg.conf.man.pre
@@ -2163,7 +2163,7 @@ General:
 .PP
 Display drivers:
 .BR apm (__drivermansuffix__),
-.\" .BR ati (__drivermansuffix__),
+.BR ati (__drivermansuffix__),
 .BR chips (__drivermansuffix__),
 .BR cirrus (__drivermansuffix__),
 .BR cyrix (__drivermansuffix__),
@@ -2172,11 +2172,12 @@ Display drivers:
 .BR glint (__drivermansuffix__),
 .BR i128 (__drivermansuffix__),
 .BR i740 (__drivermansuffix__),
-.BR i810 (__drivermansuffix__),
 .BR imstt (__drivermansuffix__),
+.BR intel (__drivermansuffix__),
 .BR mga (__drivermansuffix__),
 .BR neomagic (__drivermansuffix__),
 .BR nv (__drivermansuffix__),
+.BR openchrome (__drivermansuffix__),
 .BR r128 (__drivermansuffix__),
 .BR rendition (__drivermansuffix__),
 .BR savage (__drivermansuffix__),
@@ -2195,8 +2196,6 @@ Display drivers:
 .BR trident (__drivermansuffix__),
 .BR tseng (__drivermansuffix__),
 .BR vesa (__drivermansuffix__),
-.BR vga (__drivermansuffix__),
-.BR via (__drivermansuffix__),
 .BR vmware (__drivermansuffix__).
 .PP
 Input drivers:
-- 
1.5.6.5

From 5b9acd1d5a10e38a6fdaa99bb20efc7bcaa819a7 Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@sun.com>
Date: Mon, 1 Dec 2008 20:02:34 -0800
Subject: [PATCH] More man page updates for 1.6 release for Xorg, xorg.conf & exa man pages.

- Remove remaining references to XFree86-Misc options AllowNonLocalModInDev
  and DisableModInDev.
- Remove remaining references to grab-breaking keys & associated options.
- Update description of Ctrl-Alt-Backspace to new -retro/DontZap defaults.
- Add description of new options -modalias and -showopts.
- Update list of modules loaded by default.
- Update input driver references from keyboard to evdev & kbd.
- Update list of driver man pages to match xf86-*-* drivers with man pages.
- Add See Also section to exa man page.

and various formatting/typo/etc. fixes.

The Xorg/xorg.conf sections on input device selection could use further
updates to better match the current state of HAL-enabled configuration.

(Cherry-picked from commit 8c560422b44e012053612754430d2b87dc44ed59 and
 amended to only include changes relevant to Xorg 1.5.x branch)
---
 hw/xfree86/doc/man/Xorg.man.pre      |    8 +++---
 hw/xfree86/doc/man/xorg.conf.man.pre |   40 +++++++++++++++++++++++-----------
 hw/xfree86/exa/exa.man.pre           |   16 +++++++------
 3 files changed, 40 insertions(+), 24 deletions(-)

diff --git a/hw/xfree86/doc/man/Xorg.man.pre b/hw/xfree86/doc/man/Xorg.man.pre
index e427e51..504552c 100644
--- a/hw/xfree86/doc/man/Xorg.man.pre
+++ b/hw/xfree86/doc/man/Xorg.man.pre
@@ -133,8 +133,8 @@ specifies the Virtual Terminal device number which
 will use.  Without this option,
 .B __xservername__
 will pick the first available Virtual Terminal that it can locate.  This
-option applies only to platforms such as Linux, BSD, SVR3 and SVR4, that
-have virtual terminal support.
+option applies only to platforms that have virtual terminal support, such
+as Linux, BSD, SVR3, and SVR4.
 .TP
 .B \-allowMouseOpenFail
 Allow the server to start up even if the mouse device can't be opened
@@ -486,7 +486,7 @@ __xconfigfile__(__filemansuffix__) file option is specified, this key sequence
 deactivates any active keyboard and mouse grabs.
 .TP 8
 .B Ctrl+Alt+F1...F12
-For BSD and Linux systems with virtual terminal support, these keystroke
+For systems with virtual terminal support, these keystroke
 combinations are used to switch to virtual terminals 1 through 12,
 respectively.  This can be disabled with the
 .B DontVTSwitch
@@ -620,7 +620,7 @@ Web site
 
 .SH AUTHORS
 __xservername__ has many contributors world wide.  The names of most of them
-can be found in the documentation, CHANGELOG files in the source tree,
+can be found in the documentation, ChangeLog files in the source tree,
 and in the actual source code.
 .PP
 __xservername__ was originally based on XFree86 4.4rc2.
diff --git a/hw/xfree86/doc/man/xorg.conf.man.pre b/hw/xfree86/doc/man/xorg.conf.man.pre
index a3ab9a0..5accde6 100644
--- a/hw/xfree86/doc/man/xorg.conf.man.pre
+++ b/hw/xfree86/doc/man/xorg.conf.man.pre
@@ -803,7 +803,8 @@ __projectroot__/lib/modules/extensions
 .fi
 .RE
 .PP
-The \(lqbitmap\(rq font module is loaded automatically.
+The \(lqextmod\(rq, \(lqdbe\(rq, \(lqglx\(rq, and \(lqdri\(rq extension
+modules are loaded automatically, unless disabled with \*qDisable\*q entries.
 It is recommended
 that at very least the \(lqextmod\(rq extension module be loaded.
 If it isn't, some commonly used server extensions (like the SHAPE
@@ -863,9 +864,12 @@ or
 command line options, or if it is selected implicitly as the core pointer
 or keyboard device in the absence of such explicit references.
 The most commonly used input drivers are
-.BR keyboard (__drivermansuffix__)
+.BR evdev (__drivermansuffix__)
+on Linux systems, and
+.BR kbd (__drivermansuffix__)
 and
-.BR mouse (__drivermansuffix__).
+.BR mouse (__drivermansuffix__)
+on other platforms.
 .PP
 In the absence of an explicitly specified core input device, the first
 .B InputDevice
@@ -876,7 +880,7 @@ marked as
 is used.
 If there is no match there, the first
 .B InputDevice
-that uses the \(lqmouse\(rq (or \(lqkeyboard\(rq or \(lqkbd\(rq) driver is used.
+that uses the \(lqmouse\(rq (or \(lqkbd\(rq) driver is used.
 The final fallback is to use built\-in default configurations.
 .PP
 .B InputDevice
@@ -2159,7 +2163,7 @@ General:
 .BR cvt (__appmansuffix__),
 .BR gtf (__appmansuffix__).
 .PP
-.B Not all modules or interfaces are available on all platforms.
+.B "Not all modules or interfaces are available on all platforms."
 .PP
 Display drivers:
 .BR apm (__drivermansuffix__),
@@ -2179,11 +2183,13 @@ Display drivers:
 .BR nv (__drivermansuffix__),
 .BR openchrome (__drivermansuffix__),
 .BR r128 (__drivermansuffix__),
+.BR radeon (__drivermansuffix__),
 .BR rendition (__drivermansuffix__),
 .BR savage (__drivermansuffix__),
 .BR s3virge (__drivermansuffix__),
 .BR siliconmotion (__drivermansuffix__),
 .BR sis (__drivermansuffix__),
+.BR sisusb (__drivermansuffix__),
 .BR sunbw2 (__drivermansuffix__),
 .BR suncg14 (__drivermansuffix__),
 .BR suncg3 (__drivermansuffix__),
@@ -2192,35 +2198,43 @@ Display drivers:
 .BR sunleo (__drivermansuffix__),
 .BR suntcx (__drivermansuffix__),
 .BR tdfx (__drivermansuffix__),
-.BR tga (__drivermansuffix__),
+.\" .BR tga (__drivermansuffix__),
 .BR trident (__drivermansuffix__),
 .BR tseng (__drivermansuffix__),
 .BR vesa (__drivermansuffix__),
-.BR vmware (__drivermansuffix__).
+.BR vmware (__drivermansuffix__),
+.BR voodoo (__drivermansuffix__),
+.BR wsfb (__drivermansuffix__),
+.BR xgi (__drivermansuffix__),
+.BR xgixp (__drivermansuffix__).
 .PP
 Input drivers:
-.\" .BR acecad (__drivermansuffix__),
-.\" .BR calcomp (__drivermansuffix__),
+.BR acecad (__drivermansuffix__),
+.BR calcomp (__drivermansuffix__),
 .BR citron (__drivermansuffix__),
 .BR dmc (__drivermansuffix__),
 .BR dynapro (__drivermansuffix__),
 .BR elographics (__drivermansuffix__),
+.BR evdev (__drivermansuffix__),
 .BR fpit (__drivermansuffix__),
 .BR js_x (__drivermansuffix__),
+.BR joystick (__drivermansuffix__),
 .BR kbd (__drivermansuffix__),
-.BR keyboard (__drivermansuffix__),
-.\" .BR magictouch (__drivermansuffix__),
+.BR magictouch (__drivermansuffix__),
 .BR microtouch (__drivermansuffix__),
-.BR mouse (__drivermansuffix__),
+.BR mousedrv (__drivermansuffix__),
 .BR mutouch (__drivermansuffix__),
 .BR palmax (__drivermansuffix__),
 .BR penmount (__drivermansuffix__),
+.BR synaptics (__drivermansuffix__),
 .BR tek4957 (__drivermansuffix__),
-.\" .BR ur98 (__drivermansuffix__),
+.BR ur98 (__drivermansuffix__),
+.BR vmmouse (__drivermansuffix__),
 .BR void (__drivermansuffix__),
 .BR wacom (__drivermansuffix__).
 .PP
 Other modules and interfaces:
+.BR exa (__drivermansuffix__),
 .BR fbdevhw (__drivermansuffix__),
 .\" .BR shadowfb (__drivermansuffix__),
 .BR v4l (__drivermansuffix__).
diff --git a/hw/xfree86/exa/exa.man.pre b/hw/xfree86/exa/exa.man.pre
index 14859bc..cdd39bb 100644
--- a/hw/xfree86/exa/exa.man.pre
+++ b/hw/xfree86/exa/exa.man.pre
@@ -6,24 +6,23 @@ exa \- new 2D acceleration architecture for X.Org
 .SH DESCRIPTION
 .B EXA
 provides a simple API for video drivers to implement for 2D acceleration.  It
-is a module loaded by drivers, and is not intended to be loaded on its own.  See
-your driver's manual page for how to enable 
-.B EXA
-.
+is a module loaded by drivers, and is not intended to be loaded on its own.
+See your driver's manual page for how to enable 
+.BR EXA .
 .PP
 The
 .B EXA
 architecture is designed to make accelerating the Render extension simple and
 efficient, and results in various performance tradeoffs compared to XAA.  Some
-options are available for debugging performance issues or driver rendering
-problems.  They are not intended for general use.
+__xconfigfile__ options are available for debugging performance issues or
+driver rendering problems.  They are not intended for general use.
 .TP
 .BI "Option \*qEXANoComposite\*q \*q" boolean \*q
 Disables acceleration of the Composite operation, which is at the heart of
 the Render extension.  Not related to the Composite extension.  Default: No.
 .TP
 .BI "Option \*qEXANoUploadToScreen\*q \*q" boolean \*q
-Disables acceleration of uploading pixmap data to the freamebuffer. Default: No.
+Disables acceleration of uploading pixmap data to the framebuffer. Default: No.
 .TP
 .BI "Option \*qEXANoDownloadFromScreen\*q \*q" boolean \*q
 Disables acceleration of downloading of pixmap data from the framebuffer. 
@@ -42,5 +41,8 @@ Chooses an alternate pixmap migration heuristic, for debugging purposes.  The
 default is intended to be the best performing one for general use, though others
 may help with specific use cases.  Available options include \*qalways\*q,
 \*qgreedy\*q, and \*qsmart\*q.  Default: always.
+.SH "SEE ALSO"
+.BR __xservername__ (__appmansuffix__),
+.BR __xconfigfile__(__filemansuffix__).
 .SH AUTHORS
 Authors include: Keith Packard, Eric Anholt, Zack Rusin, and Michel D\(:anzer
-- 
1.5.6.5

From 17f9968f91e277d28fa0a124f5d46c2498a28acb Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@sun.com>
Date: Thu, 19 Feb 2009 13:57:28 -0800
Subject: [PATCH] Add Extensions section to xorg.conf man page

Extensions section was added in X11R6.8.0 and documented in the release notes:
   http://www.x.org/archive/X11R6.8.0/doc/RELNOTES2.html#3
but never made it into the man page.

Also fix a bonus typo.

Signed-off-by: Alan Coopersmith <alan.coopersmith@sun.com>
(cherry picked from commit ab61033700b5383a7a15370dd054eaa80e72e811)
---
 hw/xfree86/doc/man/xorg.conf.man.pre |   29 ++++++++++++++++++++++++++++-
 1 files changed, 28 insertions(+), 1 deletions(-)

diff --git a/hw/xfree86/doc/man/xorg.conf.man.pre b/hw/xfree86/doc/man/xorg.conf.man.pre
index 5accde6..7598ab1 100644
--- a/hw/xfree86/doc/man/xorg.conf.man.pre
+++ b/hw/xfree86/doc/man/xorg.conf.man.pre
@@ -115,6 +115,7 @@ The section names are:
 .BR "Files          " "File pathnames"
 .BR "ServerFlags    " "Server flags"
 .BR "Module         " "Dynamic module loading"
+.BR "Extensions     " "Extension enabling"
 .BR "InputDevice    " "Input device description"
 .BR "Device         " "Graphics device description"
 .BR "VideoAdaptor   " "Xv video adaptor description"
@@ -313,7 +314,7 @@ where
 .I <identifier>
 is an alphanumeric identifier,
 .I [attribute]
-is an attribute wich will be passed to the underlying FPE and
+is an attribute which will be passed to the underlying FPE and
 .I <priority>
 is a number used to order the fontfile FPEs. Examples:
 .PP
@@ -809,6 +810,32 @@ It is recommended
 that at very least the \(lqextmod\(rq extension module be loaded.
 If it isn't, some commonly used server extensions (like the SHAPE
 extension) will not be available.
+.SH "EXTENSIONS SECTION"
+The
+.B Extensions
+section is used to specify which X11 protocol extensions should be enabled
+or disabled.
+The
+.B Extensions
+section is optional, as are all of the entries that may be specified in
+it.
+.PP
+Entries in this section are listed as Option statements with the name of
+the extension as the first argument, and a boolean value as the second.
+The extension name is case\-sensitive, and matches the form shown in the output
+of \*qXorg -extension ?\*q.
+.PP
+.RS 7
+Example: the MIT-SHM extension can be disabled with the following entry:
+.PP
+.RS 4
+.nf
+.B "Section \*qExtensions\*q"
+.B "    Option \*qMIT-SHM\*q \*qDisable\*q"
+.B "EndSection"
+.fi
+.RE
+.RE
 .SH "INPUTDEVICE SECTION"
 The config file may have multiple
 .B InputDevice
-- 
1.5.6.5

